WikiWatershed/model-my-watershed
# Upgrade Front-end Dependencies

## Overview

Upgrades a number of front-end dependencies for security purposes. See commit messages for details.

Notably, Nunjucks is not upgraded, because we use [nunjucksify](https://github.com/rotundasoftware/nunjucksify) to make it work with browserify. Unfortunately, nunjucksify hasn't been worked on in 3 years, and the latest version (which we are already on) pulls in the old version of Nunjucks we're trying to move away from. This adds bloat to the JS payload, without solving the original issues, so we're going to defer that for now.

An eventual solution may be to move away from Nunjucks, or move away from Browserify, for the front-end. This will be investigated in #3140.

Connects #3100 

## Testing Instructions

* Check out this branch
* Run `./scripts/npm.sh install` to update the dependencies
* Run `./scripts/bundle.sh --debug --vendor --tests` to compile the bundle
* Run `./scripts/testem.sh` to run JavaScript tests
* Go to [:8000](http://localhost:8000/) and test the app
  - [x] Ensure there are no console errors (outside of the Google Maps API Key warning)
